### ISSUE TITLE
Single Select Radio Buttons in Workflow

### ISSUE THREAD

**Issue by BelAirIT** [SparkDevNetwork/Rock#1334]
When using Radio Button control type with a single option (e.g. Yes) the available options are None and Yes.  Clicking None defaults back to Yes after saving.  A similar behavior exists in v3.4 and v4.0.

![image](https://cloud.githubusercontent.com/assets/7409839/11795108/cb423060-a269-11e5-86c9-a24552bf15ae.png)

**Comment by DerekMangrum** [comment]
Confirmed in Rock v4.2

**Comment by azturner** [comment]
@BelAirIT "None" is the same as not selecting a value, and if a value is not specified, the default value for the attribute will be displayed which is probably "Yes". If you set the default to "None" it should work. You can also make the value required so that the "none" is not displayed and add then add your own value for none.  Closing for now, but feel free to reopen if you still think it's an issue.

**Comment by Tschrock** [comment]
IMHO This is a bug. Setting 'none' as the default value should leave all options unselected, not select the first value.

**Comment by BelAirIT** [comment]
The problem is twofold, in my mind. First, the behavior described above where clicking None and saving then defaults back to Yes. This doesn't seem to affect the attribute if you don't go back into editing it but it's still odd. The second problem is that I don't believe 'None' should be an option. Not only is it not one of the options I'd choose, it isn't the expected behavior. What @Tschrock described is what I'd expect if I chose 'None'.

All that being said, I realized that if the form sets that attribute to Required, it removes 'None' from the options (good) but still forces a default of the first value (bad).

**Comment by Tschrock** [comment]
Any word on this? We have a lot of single-selects where we don't want anything selected by default.

**Comment by azturner** [comment]
Yes, we discussed this as a team and don't really see this as an issue. When the field is configured as not being required, if we don't display the "none" option, you don't give the user the option of unselecting a value if they unintentionally clicked the single value (since they can't unselect a radio button value). Have you considered configuring it to use a dropdownlist box instead? Or using a checkbox field? I would argue that using the singleselect field with only one value, that is not required, and rendering it as a radiobutton list may not be the best use of that fieldtype.

We did notice though that if you select "None" as the default value that does not seem to persist correctly, so we will be fixing that.
